code/controllers/subsystem/vote.dm

```diff
@@ -146,6 +152,17 @@ SUBSYSTEM_DEF(vote)
 						restart = 1
 					else
 						GLOB.master_mode = .
+			if("dynamic")
+				if(SSticker.current_state > GAME_STATE_PREGAME)//Don't change the mode if the round already started.
+					return message_admins("A vote has tried to change the gamemode, but the game has already started. Aborting.")
+				GLOB.dynamic_forced_extended = choices["extended"]/voted.len > 0.5
+				if(GLOB.dynamic_forced_extended)
+					return message_admins("Dynamic extended has been voted for.")
+				var/mean = (choices["extended"]*-2+choices[PEACE2]*-2+choices[PEACE1]*-1+choices[CHAOS1]*1+choices[CHAOS2]*2)/voted.len
+				var/variance=(((-2-mean)**2)*choices["extended"]+((-2-mean)**2)*choices[PEACE2]+((-1-mean)**2)*choices[PEACE1]+((0-mean)**2)*choices[BALANCED]+((1-mean)**2)*choices[CHAOS1]+((2-mean)**2)*choices[CHAOS2])/voted.len //Sorry. Im sorry. Im trying to remove it

```

Comment:
Keep in mind this will give off a division by zero error in the unlikely occurrence of no one voting.

Reply:
https://github.com/Citadel-Station-13/Citadel-Station-13/pull/9557/commits/0a000f3165875ace0d8b89ef909cadb59170a5a2